mirrord-layer/src/lib.rs

```diff
@@ -256,82 +257,82 @@ async fn handle_daemon_message(
         DaemonMessage::Tcp(message) => {
-            tcp_mirror_handler
-                .handle_daemon_message(message)
-                .await
-                .unwrap();
+            tcp_mirror_handler.handle_daemon_message(message).await?;
+            Ok(())
         }
         DaemonMessage::FileResponse(FileResponse::Open(open_file)) => {
             debug!("DaemonMessage::OpenFileResponse {open_file:#?}!");
             debug!("file handler = {:#?}", open_file_handler);
+
             open_file_handler
-                .lock()
-                .unwrap()
+                .lock()?
                 .pop()
-                .unwrap()
-                .send(open_file.unwrap())
-                .unwrap();
+                .ok_or(LayerError::SendErrorFileResponse)?
+                .send(open_file?)
+                .map_err(|_| LayerError::SendErrorFileResponse)?;
+
+            Ok(())
         }
         DaemonMessage::FileResponse(FileResponse::Read(read_file)) => {
             // The debug message is too big if we just log it directly.
-            let _ = read_file
-                .as_ref()
+            let file_response = read_file
                 .inspect(|success| {
                     debug!("DaemonMessage::ReadFileResponse {:#?}", success.read_amount)
                 })
-                .inspect_err(|fail| error!("DaemonMessage::ReadFileResponse {:#?}", fail));
+                .inspect_err(|fail| error!("DaemonMessage::ReadFileResponse {:#?}", fail))?;
 
             read_file_handler
-                .lock()
-                .unwrap()
+                .lock()?
                 .pop()
-                .unwrap()
-                .send(read_file.unwrap())
-                .unwrap();
+                .ok_or(LayerError::SendErrorFileResponse)?
+                .send(file_response)
+                .map_err(|_| LayerError::SendErrorFileResponse)?;
+            Ok(())
         }
         DaemonMessage::FileResponse(FileResponse::Seek(seek_file)) => {
             debug!("DaemonMessage::SeekFileResponse {:#?}!", seek_file);
 
             seek_file_handler
-                .lock()
-                .unwrap()
+                .lock()?
                 .pop()
-                .unwrap()
-                .send(seek_file.unwrap())
-                .unwrap();
+                .ok_or(LayerError::SendErrorFileResponse)?
+                .send(seek_file?)
+                .map_err(|_| LayerError::SendErrorFileResponse)?;
+            Ok(())
         }
         DaemonMessage::FileResponse(FileResponse::Write(write_file)) => {
             debug!("DaemonMessage::WriteFileResponse {:#?}!", write_file);
 
             write_file_handler
-                .lock()
-                .unwrap()
+                .lock()?
                 .pop()
-                .unwrap()
-                .send(write_file.unwrap())
-                .unwrap();
+                .ok_or(LayerError::SendErrorFileResponse)?
+                .send(write_file?)
```

Comment:
why do you uwnrap the write_file? we want it to send it to the function so it can return the appropriate error..